File: ruby/testcode/fileinclusion_031.rb
require_relative 'shared'

# vuln-code-snippet start ruby_fi_open_and_eval
def handler(req)
  open(req.param('file')) { |f| eval(f.read) } # vuln-code-snippet vuln-line ruby_fi_open_and_eval
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_fi_open_and_eval

File: php/testcode/fileinclusion_031.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_fi_trim_only_multihop
function fileinclusion031(BenchmarkRequest $req): BenchmarkResponse {
    $page = trim($req->param('page'));
    include $page; // vuln-code-snippet vuln-line php_fi_trim_only_multihop
    return BenchmarkResponse::ok('Done');
}
// vuln-code-snippet end php_fi_trim_only_multihop
